src/microbe_stage/MicrobeAI.cs

```diff
@@ -652,269 +293,194 @@ private void DealWithChunks(FloatingChunk chunk, List<FloatingChunk> allChunks)
-                }
-            }
-            else if (targetChunk != null)
-            {
-                lifeState = LifeState.SCAVENGING_STATE;
-            }
-            else if (random.Next(0, 10) < 8)
+            // There's a chance to stop for a bit and letting nutrients soak in
+            // More opportunistic species will do this less.
+            if (random.Next(-Constants.MAX_SPECIES_OPPORTUNISM, Constants.MAX_SPECIES_OPPORTUNISM)
+                > SpeciesOpportunism)
             {
-                // Every 2 intervals || so
-                lifeState = LifeState.GATHERING_STATE;
-            }
-            else if (RollCheck(SpeciesActivity, 400.0f, random))
-            {
-                // Every 10 intervals || so
-                lifeState = LifeState.PLANTLIKE_STATE;
+                SetMoveSpeed(0.0f);
             }
         }
     }
 
-    // For doing run and tumble
-    private void DoRunAndTumble(Random random)
+    private float SelectRandomTargetPosition(Random random)
     {
-        // Run and tumble
-        // A biased random walk, they turn more if they are picking up less compounds.
-        // https://www.mit.edu/~kardar/teaching/projects/chemotaxis(AndreaSchmidt)/home.htm
-
-        var randAngle = previousAngle;
-        float randDist;
-
-        float compoundDifference = microbe.TotalAbsorbedCompounds.SumValues();
+        var randAngle = previousAngle + random.Next(0.1f, 1.0f);
+        previousAngle = randAngle;
+        var randDist = random.Next(200.0f, movementRadius);
+        targetPosition = new Vector3(Mathf.Cos(randAngle) * randDist, 0, Mathf.Sin(randAngle) * randDist);
+        return randAngle;
+    }
 
-        // Angle should only change if you havent picked up compounds || picked up less compounds
-        if (compoundDifference < 0 && random.Next(0, 10) < 5)
+    private void SetEngulfIfClose()
+    {
+        // Turn on engulfmode if close
+        if ((microbe.Translation - targetPosition).LengthSquared() <= 100 + (microbe.EngulfSize * 3.0f))
         {
-            randAngle = SelectRandomTargetPosition(random);
+            microbe.EngulfMode = true;
+            ticksSinceLastToggle = 0;
         }
+    }
 
-        // If last round you had 0, then have a high likelihood of turning
-        if (compoundDifference < Constants.AI_COMPOUND_BIAS && random.Next(0, 10) < 9)
+    private void LaunchToxin(Microbe target)
+    {
+        if (microbe.Hitpoints > 0 && microbe.AgentVacuoleCount > 0 &&
+            (microbe.Translation - target.Translation).LengthSquared() <= SpeciesFocus * 10.0f)
         {
-            randAngle = SelectRandomTargetPosition(random);
+            if (microbe.Compounds.GetCompoundAmount(oxytoxy) >= Constants.MINIMUM_AGENT_EMISSION_AMOUNT)
+            {
+                microbe.LookAtPoint = target.Translation;
+                microbe.QueueEmitToxin(oxytoxy);
+            }
+            else
+            {
+                DebugFlash();
```

Comment:
Shouldn't **Debug**Flash be removed from the finished code?

Reply:
yup, that was sloppy on my part.